client.go

```diff
@@ -62,7 +52,20 @@ type ClientInt interface {
 // information can be found in their respective documentation.
 // Numerous connection options may be specified by configuring a
 // and then supplying a ClientOptions type.
-type Client struct {
+
+type Client interface {
+	IsConnected() bool
+	Connect() Token
+	Disconnect(uint)
+	disconnect()
```

Comment:
Yeah the lowercase disconnect() is an internal function of the implementation rather than the interface. If we can remove it before we merge this.